VideoAPI/VideoApi.UnitTests/Controllers/ConferenceManagement/TransferParticipantTests.cs

```diff
@@ -0,0 +1,80 @@
+using Moq;
+using NUnit.Framework;
+using VideoApi.Contract.Requests;
+using VideoApi.Domain.Enums;
+using VideoApi.Services.Kinly;
+
+namespace VideoApi.UnitTests.Controllers.ConferenceManagement
+{
+    public class TransferParticipantTests : ConferenceManagementControllerTestBase
+    {
+        [Test]
+        public async Task should_move_participant_into_hearing_room()
+        {
+            var conferenceId = Guid.NewGuid();
+            var request = new TransferParticipantRequest
+            {
+                ParticipantId = Guid.NewGuid(),
+                TransferType = TransferType.Call
+            };
+            
+            var result = await Controller.TransferParticipantAsync(conferenceId, request);
+            var typedResult = (OkResult) result;
+            typedResult.Should().NotBeNull();
+            typedResult.StatusCode.Should().Be((int) HttpStatusCode.OK);
+            
+            VideoPlatformServiceMock.Verify(
+                x => x.TransferParticipantAsync(conferenceId, request.ParticipantId, RoomType.WaitingRoom,
+                    RoomType.HearingRoom), Times.Once);
+        }
+        
+        [Test]
+        public async Task should_dismiss_participant_from_hearing_room()
+        {
+            var conferenceId = Guid.NewGuid();
+            var request = new TransferParticipantRequest
+            {
+                ParticipantId = Guid.NewGuid(),
+                TransferType = TransferType.Dismiss
+            };
+            
+            var result = await Controller.TransferParticipantAsync(conferenceId, request);
+            var typedResult = (OkResult) result;
+            typedResult.Should().NotBeNull();
+            typedResult.StatusCode.Should().Be((int) HttpStatusCode.OK);
+
+            VideoPlatformServiceMock.Verify(
+                x => x.TransferParticipantAsync(conferenceId, request.ParticipantId, RoomType.HearingRoom,
+                    RoomType.WaitingRoom), Times.Once);
+        }
+        
+        [Test] public async Task should_return_kinly_status_code_on_error()
+        {
+            var conferenceId = Guid.NewGuid();
+            var request = new TransferParticipantRequest
+            {
+                ParticipantId = Guid.NewGuid(),
+                TransferType = TransferType.Call
+            };
+            var message = "Auto Test Error";
+            var response = "You're not allowed to start this hearing";
+            var statusCode = (int) HttpStatusCode.Unauthorized;
+            var exception =
+                new KinlyApiException(message, statusCode, response, null, null);
+            VideoPlatformServiceMock
+                .Setup(x => x.TransferParticipantAsync(It.IsAny<Guid>(), It.IsAny<Guid>(), It.IsAny<RoomType>(),
+                    It.IsAny<RoomType>())).ThrowsAsync(exception);
+            
+            var result = await Controller.TransferParticipantAsync(conferenceId, request);
+            var typedResult = (ObjectResult) result;
```

Comment:
we could assert type before casting for a slightly more friendly message
`result.Should().BeOfType<ObjectResult>();`

Reply:
good call!